src/psyclone/generator.py

```diff
@@ -242,33 +247,61 @@ def generate(filename, api="", kernel_paths=None, script_name=None,
     ast, invoke_info = parse(filename, api=api, invoke_name="invoke",
                              kernel_paths=kernel_paths,
                              line_length=line_length)
-    if api != "gocean1.0":
+
+    # TODO issue #1618 remove temporary flag, associated logic and Alg class
+    # plus tests (and ast variable).
+    #
+    # Temporary flag to allow optional testing of new LFRic metadata
+    # implementation (mainly that the PSyIR works with algorithm-layer
+    # code) whilst keeping the original implementation as default
+    # until it is working.
+    lfric_testing = False
```

Comment:
I suggest hoisting this to module scope so that a test can be added. Probably just one or two tests need modifying to test with and without this set to True and that will cover all of the missed lines below.

Reply:
Good idea. I was being a bit lazy I guess. I've hoisted and added tests so that we now have 100% coverage of the changes in this PR.